src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -44,7 +45,7 @@ void ImageRepository::fetchSnapshot(INvStorage& storage, const IMetadataFetcher&
 
   if (local_version > remote_version) {
     throw Uptane::SecurityException(RepositoryType::IMAGE, "Rollback attempt");
-  } else if (local_version < remote_version) {
+  } else if (local_version < remote_version || snapshot_stored_signature != snapshot.signature()) {
```

Comment:
This seems slightly more complex than `} else {`, but I guess there is reason behind it.
Are you trying to save the amount of database calls ?  Is that performance optimization worth the extra complexity ?

IMHO that totally makes sense for the timestamp role in `updateMeta` as we fetch it quite often.
But I'm not sure about the other roles, as they're only fetched when we've noticed some changes.

Reply:
>Are you trying to save the amount of database calls ? Is that performance optimization worth the extra complexity ?

I am just trying to mimic the logic of `local_version < remote_version` so the business logic for `local_version < remote_version` and `local_version = remote_version (but differ sign)`` is the same.

Reply:
sounds reasonable

Reply:
@pattivacek I agree with @vkhoroz on the condition for storing of snapshot and targets. It can be as simple as just `} else { ` which means store snapshot/targets whenever `local_version <= remote_version` since it's called only in the case if timestamp metadata are updated, and if they are updated it indicates that snapshot/targets are changed so they should be stored in DB. 
I can imagine the case when timestamp are changed while snapshot/targets are not (e.g. timestamp role key rotation) but that's OK to rewrite the metadata in this case since it happens really rarely (and never in the case of Fio).